### PR TITLE
Patch 1

### PR DESCRIPTION
**Pull request by pp-r**
upgrade to 0.9.4.2 and use php7.3

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -1,55 +1,56 @@
...
+    "packaging_format": 1,
+    "url": "https://www.galette.eu",
+    "license": "GPL-3.0-or-later",
+    "version": "0.9.4.2~ynh1",
+    "requirements": {
+        "yunohost": ">= 3.8"
+    },
+    "services": [
+        "nginx",
+        "php7-fpm",
```
Comment (ericgaspar) [inline manifest.json:13]:
```suggestion
        "php7.3-fpm",
```

File: manifest.json [hunk 2]
```diff
@@ -1,55 +1,56 @@
...
-                    "en": "Is it a public application ?"
+                    "en": "Is it a public application?"
+                },
+                "help": {
+                    "en": "If no, only connected YunoHost users will be able to interact with the app"
                 },
-                "choices": ["Yes", "No"],
-                "default": "Yes"
+                "type": "boolean",
+                "default": "0"
```
Comment (ericgaspar) [inline manifest.json:53]:
```suggestion
                "default": true
```

File: scripts/install [hunk 3]
```diff
@@ -1,117 +1,175 @@
...
+# CHECK IF THE APP CAN BE INSTALLED WITH THESE ARGS
+#=================================================
 
-sudo yunohost app setting $app admin -v $admin
+final_path=/var/www/$app
+test ! -e "$final_path" || ynh_die "This path already contains a folder"
 
-sudo yunohost app setting $app is_public -v "$is_public"
+# Normalize the url path syntax
+path_url=$(ynh_normalize_url_path "$path_url")
```
Comment (ericgaspar) [inline scripts/install:37]:
```suggestion

```
Not needed anymore...

File: scripts/install [hunk 4]
```diff
@@ -1,117 +1,175 @@
...
 
-tar xjvf ../sources/galette-$version.tar.bz2 > /dev/null
-sudo mv galette-$version/galette $final_path
-rm -r galette-$version
+#=================================================
+# INSTALL DEPENDENCIES
+#=================================================
 
-# Set permissions
+ynh_install_php --phpversion="$php_version" --package="$php_packages"
```
Comment (ericgaspar) [inline scripts/install:56]:
```suggestion

```
We don't have to install PHP

File: scripts/install [hunk 5]
```diff
@@ -1,117 +1,175 @@
...
-    sudo chmod g+rwx $final_path/data/$folder
-done
+#=================================================
+# CREATE A MYSQL DATABASE
+#=================================================
+# If your app uses a MySQL database, you can use these lines to bootstrap
+# a database, an associated user and save the password in app settings
+
+db_name=$(ynh_sanitize_dbid "$app")
+db_pwd=$(ynh_string_random 15)
```
Comment (ericgaspar) [inline scripts/install:65]: we don't need to define the db password:
`ynh_mysql_setup_db` will create a database, an associated user and a ramdom password.

File: scripts/install [hunk 6]
```diff
@@ -1,117 +1,175 @@
...
 
-# Initialize database and store mysql password for upgrade
+#=================================================
+# PHP-FPM CONFIGURATION
+#=================================================
 
-sudo yunohost app initdb $db_user -p $db_pwd
-sudo yunohost app setting $app mysqlpassword -v $db_pwd
+# Create a dedicated php-fpm config
+ynh_add_fpm_config --phpversion="$php_version"
```
Comment (ericgaspar) [inline scripts/install:99]:
```suggestion
ynh_add_fpm_config
```

File: scripts/upgrade [hunk 7]
```diff
@@ -6,4 +6,198 @@
...
+# Exit if an error occurs during the execution of the script
+ynh_abort_if_errors
+
+#=================================================
+# CHECK THE PATH
+#=================================================
+
+# Normalize the URL path syntax
+path_url=$(ynh_normalize_url_path "$path_url")
+
```
Comment (ericgaspar) [inline scripts/upgrade:78]:
```suggestion

```
not needed

File: scripts/upgrade [hunk 8]
```diff
@@ -6,4 +6,198 @@
...
+# Create a dedicated nginx config
+ynh_add_nginx_config
+
+#=================================================
+# INSTALL DEPENDENCIES
+#=================================================
+
+ynh_install_php --phpversion="$php_version" --package="$php_packages"
+
```
Comment (ericgaspar) [inline scripts/upgrade:112]:
```suggestion

```
Not needed

File: scripts/upgrade [hunk 9]
```diff
@@ -6,4 +6,198 @@
...
+# Create a system user
+ynh_system_user_create "$app"
+
+#=================================================
+# PHP-FPM CONFIGURATION
+#=================================================
+
+# Create a dedicated php-fpm config
+ynh_add_fpm_config --phpversion="$php_version"
```
Comment (ericgaspar) [inline scripts/upgrade:125]:
```suggestion
ynh_add_fpm_config
```

File: scripts/_common.sh [hunk 10]
```diff
@@ -0,0 +1,431 @@
+#!/bin/bash
+
+#=================================================
+# COMMON VARIABLES
+#=================================================
+
+# dependencies used by the app
+php_version="7.3"
+php_packages="php7.3-tidy php7.3-intl php7.3-mbstring php7.3-mysql php7.3-xml php7.3-gd php7.3-curl"
```
Comment (ericgaspar) [inline scripts/_common.sh:9]:
```suggestion
YNH_PHP_VERSION="7.3"
php_packages="php${YNH_PHP_VERSION}-tidy php${YNH_PHP_VERSION}-intl php${YNH_PHP_VERSION}-mbstring php${YNH_PHP_VERSION}-mysql php${YNH_PHP_VERSION}-xml php${YNH_PHP_VERSION}-gd php${YNH_PHP_VERSION}-curl"
```

File: scripts/_common.sh [hunk 11]
```diff
@@ -0,0 +1,431 @@
...
+Specific information for the application $app.
+$app_message
+---
+Automatic diagnosis data from YunoHost
+$(yunohost tools diagnosis | grep -B 100 "services:" | sed '/services:/d')"
+
+	# Send the email to the recipients
+	echo "$mail_message" | mail -a "Content-Type: text/plain; charset=UTF-8" -s "$mail_subject" "$recipients"
+}
+
```
Comment (ericgaspar) [inline scripts/_common.sh:63]: From here till the end you can remove the helpers. They are now official helpers

File: manifest.json [hunk 12]
```diff
@@ -1,55 +1,56 @@
 {
     "name": "Galette",
     "id": "galette",
+    "packaging_format": 1,
+    "url": "https://www.galette.eu",
+    "license": "GPL-3.0-or-later",
+    "version": "0.9.4.2~ynh1",
+    "requirements": {
+        "yunohost": ">= 3.8"
```
Comment (ericgaspar) [inline manifest.json:9]:
```suggestion
        "yunohost": ">= 3.8.1"
```
We need to have the version 3.8.1 to remove the helpers on _common.sh

**Review by ericgaspar — CHANGES_REQUESTED** [review]

**Comment by ericgaspar** [comment]
This is good to take care of *Galette* for Epiphany 😄

**Comment by pp-r** [comment]
Thanks to you because I'm not comfortable with the application packaging ... and English ... but I'm trying ...

**Comment by ericgaspar** [comment]
No problem, you can get help by looking at this example package https://github.com/YunoHost/example_ynh or an app using PHP7.3 like https://github.com/YunoHost-Apps/ttrss_ynh

**Comment by ericgaspar** [comment]
I changed the destination to a new branch `0.9.4.2` (from testing branch) so part of my suggestion are outdated)

File: conf/nginx.conf [hunk 13]
```diff
@@ -16,7 +16,7 @@ location __PATH__/ {
   try_files $uri $uri/ @galette;
   location ~ [^/]\.php(/|$) {
     fastcgi_split_path_info ^(.+?\.php)(/.*)$;
-    fastcgi_pass unix:/var/run/php7.1-fpm-__NAME__.sock;
+    fastcgi_pass unix:/var/run/php7.3-fpm-__NAME__.sock;
```
Comment (ericgaspar) [inline conf/nginx.conf:19]:
```suggestion
    fastcgi_pass unix:/var/run/php__PHPVERSION__-fpm-__NAME__.sock;
```

File: conf/php-fpm.conf [hunk 14]
```diff
@@ -30,7 +30,7 @@ group = __USER__
 ;                            specific port;
 ;   '/path/to/unix/socket' - to listen on a unix socket.
 ; Note: This value is mandatory.
-listen = /var/run/php7.1-fpm-__NAMETOCHANGE__.sock
+listen = /var/run/php7.3-fpm-__NAMETOCHANGE__.sock
```
Comment (ericgaspar) [inline conf/php-fpm.conf:33]:
```suggestion
listen = /var/run/php__PHPVERSION__-fpm-__NAMETOCHANGE__.sock
```

File: scripts/restore [hunk 15]
```diff
@@ -109,5 +109,5 @@ ynh_restore_file "/etc/logrotate.d/$app"
 # RELOAD NGINX AND PHP-FPM
 #=================================================
 
-systemctl reload php7.1-fpm
+systemctl reload php7.3-fpm
```
Comment (ericgaspar) [inline scripts/restore:112]:
```suggestion
systemctl reload php$phpversion-fpm
```

**Review by ericgaspar — CHANGES_REQUESTED** [review]

**Review by ericgaspar — APPROVED** [review]

**Pull request merged**